src/overlays/actors/ovl_En_Kanban/z_en_kanban.c

```diff
@@ -25,15 +83,777 @@ const ActorInit En_Kanban_InitVars = {
+                        piece->pieceType = PIECE_LOWER_LEFT;
+                    } else if (piece->partFlags & PART_UPPER_RIGHT) {
+                        piece->pieceType = PIECE_UPPER_RIGHT;
+                    } else if (piece->partFlags & PART_RIGHT_UPPER) {
+                        piece->pieceType = PIECE_RIGHT_UPPER;
+                    } else if (piece->partFlags & PART_RIGHT_LOWER) {
+                        piece->pieceType = PIECE_RIGHT_LOWER;
+                    } else if (piece->partFlags & PART_LOWER_RIGHT) {
+                        piece->pieceType = PIECE_LOWER_RIGHT;
+                    } else if (piece->partFlags & PART_POST_UPPER) {
+                        piece->pieceType = PIECE_POST_UPPER;
+                    } else if (piece->partFlags & PART_POST_LOWER) {
+                        piece->pieceType = PIECE_POST_LOWER;
+                    } else {
+                        piece->pieceType = PIECE_OTHER;
+                    }
+                    if (piece->pieceType == 100) {
+                        piece->pieceType = PIECE_WHOLE_SIGN;
+                    }
+
+                    Matrix_RotateY((this->actor.shape.rot.y / (f32)0x8000) * M_PI, MTXMODE_NEW);
+                    Matrix_MultVec3f(&sPieceOffsets[piece->pieceType], &offset);
+                    piece->actor.posRot.pos.x += offset.x;
+                    piece->actor.posRot.pos.y += offset.y;
+                    piece->actor.posRot.pos.z += offset.z;
+                    piece->offset.x = -sPieceOffsets[piece->pieceType].x / this->actor.scale.x;
+                    piece->offset.y = -sPieceOffsets[piece->pieceType].y / this->actor.scale.x;
+                    piece->offset.z = -sPieceOffsets[piece->pieceType].z / this->actor.scale.x;
+                    piece->pieceWidth = sPieceSizes[piece->pieceType].x;
+                    piece->pieceHeight = sPieceSizes[piece->pieceType].y;
+                    piece->actor.gravity = -1.0f;
+                    piece->actionState = ENKANBAN_AIR;
+                    piece->actor.posRot.rot.y =
+                        (s16)Rand_CenteredFloat(0x3000) + this->actor.yawTowardsLink + 0x8000;
+                    piece->actor.velocity.y = Rand_ZeroFloat(2.0f) + 3.0f;
+                    piece->actor.speedXZ = Rand_ZeroFloat(2.0f) + 3.0f;
+                    if (piece->partCount >= 4) {
+                        piece->bounceX = (s16)Rand_ZeroFloat(10.0f) + 6;
+                        piece->bounceZ = (s16)Rand_ZeroFloat(10.0f) + 6;
+                    } else {
+                        piece->bounceX = (s16)Rand_ZeroFloat(7.0f) + 3;
+                        piece->bounceZ = (s16)Rand_ZeroFloat(7.0f) + 3;
+                    }
+                    piece->spinVel.y = Rand_CenteredFloat(0x1800);
+                    if (Rand_ZeroOne() < 0.5f) {
+                        piece->direction = 1;
+                    } else {
+                        piece->direction = -1;
+                    }
```

Comment:
ternary might look nice here, but its fine either way

Reply:
Surprisingly it doesn't match as a ternary.